corehq/apps/users/models.py

```diff
@@ -806,13 +814,28 @@ def wrap(cls, data, should_save=False):
             should_save = True
 
         data = cls.migrate_eula(data)
+        data = cls.migrate_device_ids(data)
 
         couch_user = super(CouchUser, cls).wrap(data)
         if should_save:
             couch_user.save()
 
         return couch_user
 
+    @classmethod
+    def migrate_device_ids(cls, data):
+        if (
+            data.get('device_ids') and
+            isinstance(data['device_ids'][0], basestring)
+        ):
+            # device_ids property was only set once, on creation, with a single string, so we only need to worry
+            # about one item in the list
+            data['device_ids'] = [{
+                'device_id': data['device_ids'][0],
```

Comment:
What I didn't understand until I tested it is that the data that is passed to wrap is loaded from JSON. So the error when trying to instantiate a DeviceIdLastUsed was hiding the error I got when trying to pass that DeviceIdLastUsed instance to `super().wrap()` (in line 819).

The one below, in the `create()` method works because it's building a Python object, and `commcare_user.created_on` has already been set to a datetime instance by `CouchUser.create()`.